quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -305,6 +305,11 @@ fn previous_position_for_offset(offset: i64) -> Position {
     }
 }
 
+/// Checks if connecting with the given parameters works.
+pub(super) fn check(client_params: serde_json::Value) -> anyhow::Result<()> {
+    create_consumer(None, client_params).map(drop)
```

Comment:
My understanding is that your intent is `.map(|_| ())`.

I think the most readable code for that is simply:

```suggestion
    create_consumer(None, client_params)?;
    Ok(())
```